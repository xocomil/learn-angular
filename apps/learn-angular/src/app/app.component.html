<fieldset>
  <label><input type="checkbox" [(ngModel)]="todosList[0].done"> {{todosList[0].description}} ({{todosList[0].done}})</label>
  <label><input type="checkbox" [(ngModel)]="todosList[1].done"> {{todosList[1].description}} ({{todosList[1].done}})</label>
</fieldset>
