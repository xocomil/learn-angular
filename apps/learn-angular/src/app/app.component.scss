.main-display {
  width: 80%;
  margin: 2rem auto auto;
}

.new-todo-footer {
  display: flex;
  padding: 0 1rem;

  .todo-description {
    flex-grow: 1;
  }

  .add-btn {
    max-height: 2.5rem;
    margin-top: 0.7rem;
  }
}

.clear-all-btn {
  margin: 0.5rem;
}

.delete-btn {
  color: red;
}
