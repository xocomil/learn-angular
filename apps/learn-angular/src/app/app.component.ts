import { Component } from '@angular/core';

@Component({
  selector: 'learn-angular-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent {
  title = 'learn-angular';
}
