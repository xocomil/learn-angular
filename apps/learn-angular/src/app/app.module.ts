import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { MatCardModule } from '@angular/material/card';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserAnimationsModule, BrowserModule, FormsModule, MatCardModule, MatCheckboxModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
