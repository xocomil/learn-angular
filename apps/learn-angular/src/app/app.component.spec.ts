import { createComponentFactory } from '@ngneat/spectator/jest';
import { Store } from '@ngrx/store';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  const createComponent = createComponentFactory({
    component: AppComponent,
    mocks: [Store],
  });

  it('should create the app', () => {
    const spectator = createComponent();

    expect(spectator).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const spectator = createComponent();

    expect(spectator.fixture).toMatchSnapshot();
  });
});
