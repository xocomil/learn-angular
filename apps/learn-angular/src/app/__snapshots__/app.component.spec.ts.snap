// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match the snapshot 1`] = `
<learn-angular-root
  __ngContext__={[Function Number]}
  title={[Function String]}
>
  <div
    class="main-display"
  >
    <learn-angular-todo>
      <mat-card
        class="mat-card mat-focus-indicator _mat-animation-noopable"
      >
        <mat-card-title
          class="mat-card-title"
        >
          Things to do
        </mat-card-title>
        <mat-card-content
          class="mat-card-content"
        >
          <section>
          </section>
        </mat-card-content>
        <mat-card-footer
          class="mat-card-footer"
        >
          <form
            class="ng-untouched ng-pristine ng-valid"
            novalidate=""
          >
            <div
              class="new-todo-footer"
            >
              <mat-form-field
                appearance="fill"
                class="mat-form-field todo-description mat-primary mat-form-field-type-mat-input mat-form-field-appearance-fill mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
              >
                <div
                  class="mat-form-field-wrapper"
                >
                  <div
                    class="mat-form-field-flex"
                  >
                    <div
                      class="mat-form-field-infix"
                    >
                      <input
                        aria-invalid="false"
                        aria-required="false"
                        class="mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-valid cdk-text-field-autofill-monitored"
                        data-placeholder="Something to do"
                        matinput=""
                        name="newTodoDescription"
                        placeholder="Something to do"
                      />
                      <span
                        class="mat-form-field-label-wrapper"
                      >
                        <label
                          class="mat-form-field-label mat-empty mat-form-field-empty"
                        >
                          <mat-label>
                            Something you need to do
                          </mat-label>
                        </label>
                      </span>
                    </div>
                  </div>
                  <div
                    class="mat-form-field-underline"
                  >
                    <span
                      class="mat-form-field-ripple"
                    />
                  </div>
                  <div
                    class="mat-form-field-subscript-wrapper"
                  >
                    <div
                      class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                      style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                    >
                      <div
                        class="mat-form-field-hint-spacer"
                      />
                    </div>
                  </div>
                </div>
              </mat-form-field>
              <button
                class="mat-focus-indicator add-btn mat-raised-button mat-button-base mat-primary _mat-animation-noopable mat-button-disabled"
                color="primary"
                disabled="true"
                mat-raised-button=""
                matsuffix=""
                type="submit"
              >
                <span
                  class="mat-button-wrapper"
                >
                  Add
                </span>
                <span
                  class="mat-ripple mat-button-ripple"
                  matripple=""
                />
                <span
                  class="mat-button-focus-overlay"
                />
              </button>
            </div>
          </form>
          <button
            class="mat-focus-indicator clear-all-btn mat-raised-button mat-button-base _mat-animation-noopable"
            mat-raised-button=""
            type="button"
          >
            <span
              class="mat-button-wrapper"
            >
              Clear All
            </span>
            <span
              class="mat-ripple mat-button-ripple"
              matripple=""
            />
            <span
              class="mat-button-focus-overlay"
            />
          </button>
        </mat-card-footer>
      </mat-card>
    </learn-angular-todo>
    <learn-angular-todo>
      <mat-card
        class="mat-card mat-focus-indicator _mat-animation-noopable"
      >
        <mat-card-title
          class="mat-card-title"
        >
          Things to do
        </mat-card-title>
        <mat-card-content
          class="mat-card-content"
        >
          <section>
          </section>
        </mat-card-content>
        <mat-card-footer
          class="mat-card-footer"
        >
          <form
            class="ng-untouched ng-pristine ng-valid"
            novalidate=""
          >
            <div
              class="new-todo-footer"
            >
              <mat-form-field
                appearance="fill"
                class="mat-form-field todo-description mat-primary mat-form-field-type-mat-input mat-form-field-appearance-fill mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
              >
                <div
                  class="mat-form-field-wrapper"
                >
                  <div
                    class="mat-form-field-flex"
                  >
                    <div
                      class="mat-form-field-infix"
                    >
                      <input
                        aria-invalid="false"
                        aria-required="false"
                        class="mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-valid cdk-text-field-autofill-monitored"
                        data-placeholder="Something to do"
                        matinput=""
                        name="newTodoDescription"
                        placeholder="Something to do"
                      />
                      <span
                        class="mat-form-field-label-wrapper"
                      >
                        <label
                          class="mat-form-field-label mat-empty mat-form-field-empty"
                        >
                          <mat-label>
                            Something you need to do
                          </mat-label>
                        </label>
                      </span>
                    </div>
                  </div>
                  <div
                    class="mat-form-field-underline"
                  >
                    <span
                      class="mat-form-field-ripple"
                    />
                  </div>
                  <div
                    class="mat-form-field-subscript-wrapper"
                  >
                    <div
                      class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                      style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                    >
                      <div
                        class="mat-form-field-hint-spacer"
                      />
                    </div>
                  </div>
                </div>
              </mat-form-field>
              <button
                class="mat-focus-indicator add-btn mat-raised-button mat-button-base mat-primary _mat-animation-noopable mat-button-disabled"
                color="primary"
                disabled="true"
                mat-raised-button=""
                matsuffix=""
                type="submit"
              >
                <span
                  class="mat-button-wrapper"
                >
                  Add
                </span>
                <span
                  class="mat-ripple mat-button-ripple"
                  matripple=""
                />
                <span
                  class="mat-button-focus-overlay"
                />
              </button>
            </div>
          </form>
          <button
            class="mat-focus-indicator clear-all-btn mat-raised-button mat-button-base _mat-animation-noopable"
            mat-raised-button=""
            type="button"
          >
            <span
              class="mat-button-wrapper"
            >
              Clear All
            </span>
            <span
              class="mat-ripple mat-button-ripple"
              matripple=""
            />
            <span
              class="mat-button-focus-overlay"
            />
          </button>
        </mat-card-footer>
      </mat-card>
    </learn-angular-todo>
  </div>
</learn-angular-root>
`;
